runtime/io.c

```diff
@@ -306,6 +332,7 @@ CAMLexport int caml_getblock(struct channel *channel, char *p, intnat len)
   } else {
     nread = caml_read_fd(channel->fd, channel->flags, channel->buff,
                          channel->end - channel->buff);
+    if (nread == Io_interrupted) goto again;
```

Comment:
Is it intentional that there is no corresponding check for `nread == 0` here to check for an EOF condition, as there is with `caml_read_fd` a few hunk above?

Reply:
That is odd! I'm trying to preserve existing behaviour here, and it looks like `caml_refill` raises end_of_file on a zero read but `caml_getblock` does not. The only user of getblock in the stdlib (md5.c) has an explicit == 0 check after the call.